mig/shared/install.py

```diff
@@ -591,7 +591,7 @@ def generate_confs(
     user_dict = _prepare(options, **expanded)
     _writefiles(options, user_dict)
     _instructions(options, user_dict)
-    return options
+    return (options, user_dict)
```

Comment:
You're spot-on right in your analysis - it strictly doesn't really. It was just that I started out extending the `test_mig_install_generateconfs.py` to also do inline value checking until I realized you already added checks on the actual file contents in `test_mig_share_install.py`. 
Still, it's a minor change and I thought we may still want to have access to the generated content e.g. to also validate that what we generate is what we write or other more dynamic uses of the conf generator.

Reply:
Hmm ok - couple of things. First is since this didn't really do anything in this PR it actually made the diff somewhat noisy to read and ultimately this would have been better introduced at the point we actually needed it :) for the record I had been trying to make user_dict opaque, and this starts to expose it in a way I had been trying to avoid which is why I'd pushed back. Again, I might be wrong about being able to avoid accessing it, but this may end up with another commit that undoes this later which feels a bit like churn. Anyway, didn't really change much in the end.